functorch/csrc/DynamicLayer.h

```diff
@@ -39,12 +41,14 @@ struct TORCH_API DynamicLayer {
   // Honestly these should be a union or some extendable metadata class.
   // Not doing that for now because I don't think we'll use this mechanism for very long.
   optional<int64_t> batchSize_;
+  optional<std::string> randomness_;
```

Comment:
This is a string in Python land but the usual way we represent string-like things in C++ are enums. enums save memory and dynamic allocations and are faster.